cartridge/lua-api/proxy.lua

```diff
@@ -0,0 +1,67 @@
+local yaml = require('yaml')
+local checks = require('checks')
+local errors = require('errors')
+local membership = require('membership')
+
+local pool = require('cartridge.pool')
+local confapplier = require('cartridge.confapplier')
+
+local vars = require('cartridge.vars').new('cartridge.lua-api.proxy')
+
+vars:new('destination')
+
+local function call(function_name, ...)
+    checks('string')
+
+    if confapplier.get_state() ~= 'Unconfigured' then
+        return nil
+    end
+
+    if vars.destination ~= nil then
+        local member = membership.get_member(vars.destination)
+        if member ~= nil and member.status == 'alive' then
+            goto call
+        end
+    end
+
+    for _, member in membership.pairs() do
+        if member.status ~= 'alive'
+        or member.payload.uuid == nil
+        then
+            goto continue
+        end
+
+        -- For the sake of destination determinism choose
+        -- the member with the lowest uri (lexicographically)
+        if vars.destination == nil
+        or member.uri < vars.destination
+        then
+            vars.destination = member.uri
+        end
+
+        ::continue::
+    end
+
+    if vars.destination == nil then
+        return nil
+    end
+
+::call::
+    local conn = pool.connect(vars.destination, {wait_connected = false})
+    local ret, err = errors.netbox_eval(conn, [[
```

Comment:
It it possible that you will perform a call before connection will be established?

Reply:
Yes, the connection internal state machine handles it fine. Moreover, it's the only valid way to avoid long timeouts (like in #1274)